src/core/device.c

```diff
@@ -915,20 +915,19 @@ static int device_dispatch_io(sd_device_monitor *monitor, sd_device *dev, void *
                 return 0;
         }
 
-        if (action == DEVICE_ACTION_CHANGE)
+        if (!IN_SET(action, DEVICE_ACTION_ADD, DEVICE_ACTION_REMOVE, DEVICE_ACTION_MOVE))
                 device_propagate_reload_by_sysfs(m, sysfs);
```

Comment:
This is the only-line with non-whitespace change. Effectively this enabled reload propagation also for BIND, UNBIND, ONLINE, OFFLINE. Sounds reasonable.